enclave-runtime/src/lib.rs

```diff
@@ -650,30 +676,33 @@ where
 		// execute indirect calls, incl. shielding and unshielding
 		match scan_block_for_relevant_xt(&signed_block.block, on_chain_ocall_api) {
 			// push shield funds to opaque calls
-			Ok(c) => calls.extend(c.into_iter()),
+			Ok(_) => {}, //calls.extend(c.into_iter()),
 			Err(_) => error!("Error executing relevant extrinsics"),
 		};
 
-		// compose indirect block confirmation
-		let xt_block = [TEEREX_MODULE, BLOCK_CONFIRMED];
-		let genesis_hash = validator.genesis_hash(validator.num_relays()).unwrap();
-		let block_hash = signed_block.block.header().hash();
-		let prev_state_hash = signed_block.block.header().parent_hash();
-		calls.push(OpaqueCall::from_tuple(&(
-			xt_block,
-			genesis_hash,
-			block_hash,
-			prev_state_hash.encode(),
-		)));
+		// FIXME: do we need a call confirmation for indirect calls?
+		// // compose indirect call confirmation
+		// let xt_block = [TEEREX_MODULE, BLOCK_CONFIRMED];
+		// let genesis_hash = validator.genesis_hash(validator.num_relays()).unwrap();
+		// let block_hash = signed_block.block.header().hash();
+		// let prev_state_hash = signed_block.block.header().parent_hash(); // this is not the previous state hash!
+		// calls.push(OpaqueCall::from_tuple(&(
+		// 	xt_block,
+		// 	genesis_hash,
+		// 	block_hash,
+		// 	prev_state_hash.encode(),
+		// )));
```

Comment:
@haerdib so when you say it was implemented properly, does that mean we don't comment this out, and leave it as it was before?

Reply:
Yes, I think so. This should be solved in a separate PR. Sorry for the confusion..

But could you adapt the comment of the function functionality? I.e.:
https://github.com/integritee-network/worker/pull/454/files#diff-69e563b2b2f86d2b47c6875cf59cc19cae787dfa172b746e37694d7d58302b9cR614-R618
```rust
/// Sync parentchain blocks to the light-client:
/// * iterates over parentchain blocks and scans for relevant extrinsics
/// * validates and execute those extrinsics (containing indirect calls), mutating state
/// * send `confirm_call` xt's of the `Stf` functions executed due to indirect invocation to the parentchain
```
to something like:
```rust
/// Sync parentchain blocks to the light-client:
/// * iterates over parentchain blocks and scans for relevant extrinsics
/// * validates and execute those extrinsics (containing indirect calls), mutating state
/// * send `confirm_call` xt's of the executed unshielding calls
/// * sends `confirm_blocks` xt's for every synced parentchain block
```

Reply:
done 👍